impeller/entity/save_layer_utils.cc

```diff
@@ -64,8 +71,27 @@ std::optional<Rect> ComputeSaveLayerCoverage(
       return source_coverage_limit;
     }
 
-    return coverage.TransformBounds(effect_transform)
-        .Intersection(source_coverage_limit.value());
+    // Trimming the content coverage by the coverage limit can reduce memory
+    // coverage. limit can reduce memory bandwith. But in cases where there are
+    // animated matrix filters, such as in the framework's zoom transition, the
+    // changing scale values continually change the source_coverage_limit.
+    // Intersecting the source_coverage_limit with the coverage may result in
+    // slightly different texture sizes each frame of the animation. This leads
+    // to non-optimal allocation patterns as differently sized textures cannot
+    // be reused. Hence the following herustic: If the coverage is within a
+    // semi-arbitrary percentage of the intersected coverage, then just use the
+    // transformed coverage. In other cases, use the intersection.
```

Comment:
What about going further and using surfaces that are rounded up to a tile size or some other quantization size? Would we be able to reuse the surfaces?

Some thought would need to be taken with kClamp/repeat/mirror tile modes, though.

Reply:
OOh, that is a good idea.

Reply:
Can the clamp/repeat modes restrict themselves to a coordinate sample not on the edge?

Reply:
We control the input UVs for all of our image filters, so should be able to easily (tm) restrict the input range.